### DISCUSSION TITLE
MathJax Not Working For In-Line Math

### DISCUSSION THREAD

**Discussion by azorbz** [community/community#56065] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am developing a maths-resources website and am using MathJax to render the maths. I implemented a solution I saw online and it works perfectly when the files are stored on my machine. However, when I upload my files to GitHub and have it as a Pages site, the in-line maths stops working on PC. When the site is viewed on mobile, the in-line maths works fine, but the in-line maths does not work when viewing it on the PC (the other maths renders fine) after testing in multiple browsers. 

I am using this script in my HTML pages.
`<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>`
and `<script>MathJax = {tex: {inlineMath: [["$", "$"]]}}</script>`.

An example page is [Normed Spaces](https://azorbz.github.io/atlasmaths/normed-spaces-21-05-2023.html).

When viewed while my files are stored locally:

![photo1](https://github.com/community/community/assets/38183961/93971f86-a272-4790-8798-4cff9f9ce625)

When viewed via GitHub Pages (Desktop):

![photo2](https://github.com/community/community/assets/38183961/25f61429-660d-4720-b003-dd3a33c180c3)

When viewed via GitHub Pages (mobile):

![photo3](https://github.com/community/community/assets/38183961/31e51676-ca63-4ac1-9fac-2521dff28e4f)

**Comment by sh1boot** [comment]
I have this too.  Under desktop Firefox it seems to only work locally.  Safari desktop works fine.  Firefox Android works fine.  Firefox on OSX breaks but only when going out to the Pages site.

**Comment by LiteBrite82** [comment]
That is so weird... 😕. I followed the link you provided and it shows up as intended for me. MacBook Pro M1, Google Chrome.
What does your inspect terminal say?

**Reply by LiteBrite82** [reply]
Its good in safari for me too

**Reply by LiteBrite82** [reply]
Browser compatibility?: https://docs.mathjax.org/en/latest/misc/faq.html#faq-ie-support

**Reply by LiteBrite82** [reply]
wait... now its not... am I crazy? I swear it looked right when I last checked. Now I see the errors...

**Reply by LiteBrite82** [reply]
And now its back to normal after page refresh.....

**Reply by LiteBrite82** [reply]
I wonder if its an API thing... can you use "await" in your api calls... unless its just used as a script... check out this article: https://stackoverflow.com/questions/7308908/waiting-for-dynamically-loaded-script

**Comment by dpvc** [comment]
Your configuration script needs to come BEFORE the script that loads `tex-mml-chtml.js`, otherwise MathJax can be loaded (from cache for example) before the configuration is loaded, and so your inline math delimiters won't be configured.  Reverse the two scripts and you should be fine.

**Comment by github-actions** [comment]
🕒 **Discussion Activity Reminder** 🕒

This Discussion has been labeled as dormant by an automated system for having no activity in the last 60 days. Please consider one the following actions:

1️⃣ Close as Out of Date: If the topic is no longer relevant, close the Discussion as `out of date` at the bottom of the page.

2️⃣ Provide More Information: Share additional details or context — or let the community know if you've found a solution on your own.

3️⃣ Mark a Reply as Answer: If your question has been answered by a reply, mark the most helpful reply as the solution.

Note: This dormant notification will only apply to Discussions with the `Question` label. To learn more, see our [recent announcement](https://github.com/orgs/community/discussions/70478).

Thank you for helping bring this Discussion to a resolution! 💬
